### ISSUE TITLE
[bug] enigma crashes when clicking outside of search box

### ISSUE THREAD

**Issue by NoahvdAa** [QuiltMC/enigma#12]
When leaving the search box open, enigma will crash when you try to click on it again.

**Reproduction steps**

1. Open Enigma and select a class.
2. Press CTRL + F to open the search bar.
3. Click on the contents of the class, above the search box.
4. Click on the search bar again, and you will see the "Enigma has crashed! =(" message pop up. :(

Might be worth noting that after this happens, the search bar stays on top of all windows, even windows owned by other programs.

The following exception is thrown:
```
java.awt.IllegalComponentStateException: component must be showing on the screen to determine its location
	at java.desktop/java.awt.Component.getLocationOnScreen_NoTreeLock(Component.java:2113)
	at java.desktop/java.awt.Component.getLocationOnScreen(Component.java:2087)
	at java.desktop/javax.swing.plaf.basic.BasicToolBarUI$Handler.mouseReleased(BasicToolBarUI.java:1352)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389)
	at java.desktop/java.awt.Component.processEvent(Component.java:6391)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by ByMartrixx** [comment]
I can't seem to reproduce this, the search bar disappears when clicking outside it. Could you provide more information about the issue, like your java version and os?

**Comment by NoahvdAa** [comment]
I'm on macOS 11.4.
`java -version` output:
```
openjdk version "17.0.1" 2021-10-19
OpenJDK Runtime Environment Temurin-17.0.1+12 (build 17.0.1+12)
OpenJDK 64-Bit Server VM Temurin-17.0.1+12 (build 17.0.1+12, mixed mode, sharing)
```

Screencap of the issue:

https://user-images.githubusercontent.com/44026893/160696600-01133342-27c5-4465-9b9a-6d50e226f493.mp4

**Comment by ByMartrixx** [comment]
Does the search dialog hide if you press escape after opening it?

**Comment by NoahvdAa** [comment]
If I don't do anything else after opening it, yes. If I click on the code escape does nothing.

**Comment by ByMartrixx** [comment]
Could you try changing the enigma version to `1.1.3-SNAPSHOT`?

**Comment by NoahvdAa** [comment]
Still crashes on 1.1.3-SNAPSHOT, exact same exception.

**Comment by NoahvdAa** [comment]
fwiw, this does *not* happen on Fabric's fork of enigma

**Comment by ix0rai** [comment]
@NoahvdAa could you test on the latest version? there hasn't been a fix targeted specifically at this issue, but there's been enough changes in the GUI code that it may have been fixed.

**Comment by ix0rai** [comment]
@NoahvdAa should be fixed in `1.7.2`, please test!
